detections/endpoint/spoolsv_writing_a_dll___sysmon.yml

```diff
@@ -0,0 +1,72 @@
+name: Spoolsv Writing a DLL - Sysmon
+id: 347fd388-da87-11eb-836d-acde48001122
+version: 1
+date: '2021-07-01'
+author: Splunk Threat Research Team
+type: batch
+datamodel:
+- Endpoint
+description: The following analytic identifies a `.dll` being written by `spoolsv.exe`.
+  This was identified during our testing of CVE-2021-1675 or PrintNightmare. Typically,
+  this is not normal behavior for `spoolsv.exe` to write a `.dll`. Current POC code
+  used will write the suspicious DLL to disk within a path of `\spool\drivers\x64\`.
+  During triage, isolate the endpoint and review for source of exploitation. Capture
+  any additional file modification events.
+search: '`sysmon` EventID=11 process_name=spoolsv.exe file_path="*\\spool\\drivers\\x64\\*"
+  file_name=*.dll | stats count min(_time) as firstTime max(_time) as lastTime by
+  dest, UserID, process_name, file_path, file_name, TargetFilename | `security_content_ctime(firstTime)`
+  | `security_content_ctime(lastTime)` | `spoolsv_writing_a_dll___sysmon_filter`'
+how_to_implement: To successfully implement this search, you need to be ingesting
+  logs with the process name, parent process, and command-line executions from your
+  endpoints. If you are using Sysmon, you must have at least version 6.0.4 of the
+  Sysmon TA. Tune and filter known instances where renamed rundll32.exe may be used.
+known_false_positives: Limited false positives. Filter as needed.
+references:
+- https://github.com/cube0x0/impacket/commit/73b9466c17761384ece11e1028ec6689abad6818
+- https://blog.truesec.com/2021/06/30/fix-for-printnightmare-cve-2021-1675-exploit-to-keep-your-print-servers-running-while-a-patch-is-not-available/
+- https://blog.truesec.com/2021/06/30/exploitable-critical-rce-vulnerability-allows-regular-users-to-fully-compromise-active-directory-printnightmare-cve-2021-1675/
+- https://www.reddit.com/r/msp/comments/ob6y02/critical_vulnerability_printnightmare_exposes
+tags:
+  analytic_story:
+  - PrintNightmare CVE-2021-1675
+  dataset:
+  - https://media.githubusercontent.com/media/splunk/attack_data/master/datasets/attack_techniques/T1547.012/printnightmare/windows-sysmon.log
+  kill_chain_phases:
+  - Exploitation
+  mitre_attack_id:
+  - T1547.012
+  product:
+  - Splunk Enterprise
+  - Splunk Enterprise Security
+  - Splunk Cloud
+  required_fields:
+  - _time
+  - dest
+  - UserID
+  - process_name
+  - file_path
+  - file_name
+  - TargetFilename
+  security_domain: endpoint
+  impact: 80
+  confidence: 90
+  risk_score: 72
+  context:
+  - Source: Endpoint
+  - Stage: Privilege Escalation, Defense Evasion
+  message: $process_name$ has been identified writing dll's to $file_path$ on endpoint
+    $dest$. This behavior is suspicious and related to PrintNightmare.
+  observable:
+  - name: dest
+    type: Endpoint
+    role:
+    - Victim
+  - name: process_name
+    type: Process Name
+    role:
```

Comment:
I would change this to process_id .. but you should then bring that into the results in the stats count.